vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ItemClickEvent.java

```diff
@@ -90,6 +96,18 @@ public ItemClickEvent(Grid<T> source, boolean fromClient,
         super(source, fromClient, screenX, screenY, clientX, clientY,
                 clickCount, button, ctrlKey, shiftKey, altKey, metaKey);
         item = source.getDataCommunicator().getKeyMapper().get(itemKey);
+        // id of column should be set for this to work
+        if(columnKey != null && !columnKey.isEmpty()) {
+            column = source.getColumns().stream().filter(col -> col.getId().isPresent() && columnKey.equals(col.getId().get())).findFirst().orElse(null);
+        }
+        // if id is not set, use internal flow ids
+        else if(flowColumnKey != null && !flowColumnKey.isEmpty() && flowColumnKey.matches("col\\d+")) {
```

Comment:
Using the `_flowId` should be a reliable way to retrieve the column instance at server-side, so it should be the only method to use, instead of being a backup-solution in case the user didn't set the id-attributes.

Grid creates these unique ids for all columns automatically, and saves the references in `idToColumnMap`: https://github.com/vaadin/vaadin-grid-flow/blob/61b6c67afc25bc39511ea1e786d8c322fb10471f/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java#L1605-L1606

This `idToColumnMap` should be used here to retrieve the `Column` instance based on the column id. The problem is that it's private. I suggest adding a package-protected method to Grid:
`Column<T> getColumnByInternalId(String id)`
to retrieve values from that map.

Note: It's important to distinguish the terms "id" and "key" in this context. Ids are part of the Grid's internal implementation, and keys are the identifiers that the user can define. Also note that I'm not talking about HTML id-attribute (`setId`) here. :)

Another note: `_flowId` is the name of the property to set for the `<vaadin-grid-column>` web component. It's named with "flow"-prefix just to point that it's part of the Flow component implementation, and to avoid conflicting with any of the web component's own properties. At server-side, the "flow"-part is irrelevant in naming, as it's obviously running Flow. :)

Reply:
ok, that I did not know, thanks :) code updated to follow the above